index.js

```diff
@@ -31,7 +31,10 @@ module.exports = {
         'jest/no-jasmine-globals': 'warn',
         'jest/no-test-prefixes': 'error',
         'jest/valid-describe': 'error',
-        'jest/valid-expect': 'error',
+        'jest/valid-expect': [
+          'error',
+          { alwaysAwait: true, ignoreInPromise: true },
```

Comment:
If those are defaults for the preset, they should probably also be defaults for the rule?

Reply:
Also, might want to default `alwaysAwait` to `false` because
```js
test('1', () => expect(doStuff(1)).resolves.toEqual(11));
test('2', () => expect(doStuff(2)).resolves.toEqual(12));
```
probably has some use cases where a whole lot of tests look nicer as one liners, I imagine.

Reply:
Should default to `false` yes

Reply:
> If those are defaults for the preset, they should probably also be defaults for the rule?

@jeysal you are right. I think setting default values for the rule and removing them from the preset makes more sense, right? Also I prefer to keep alwaysAwait: `true` & ignoreInPromise: `false` by default. I think that would be the most desired behaviour in most cases. What do you think about that?

> Also, might want to default `alwaysAwait` to `false` because
> 
> ```js
> test('1', () => expect(doStuff(1)).resolves.toEqual(11));
> test('2', () => expect(doStuff(2)).resolves.toEqual(12));
> ```
> 
> probably has some use cases where a whole lot of tests look nicer as one liners, I imagine.

Actually having alwaysAwait enabled doesn't conflict with one liners. See: https://github.com/jest-community/eslint-plugin-jest/pull/255/files#diff-a737bbc15714dae2905b79ec37ea79cdR69 

alwaysAwait just forces to use `await` instead of `return`.

Reply:
Hmm, I agree that `await` is better than `return` in a test function with a block body, just not sure I'd impose this preference on everyone by default. Why `ignoreInPromise: false`, is it too lenient and thus likely to produce false negatives?

Reply:
> Why ignoreInPromise: false, is it too lenient and thus likely to produce false negatives?

I think so. In most cases devs would return or await `Promise.all([expect1, expect])` statements. I rarely saw Promise.all(...).then(...) usages in the tests and even then reminding devs to return or await Promises (that contains async assertions) could be more useful in most cases. 

But of course this is very opinionated, just talking from my experience. I'm ok with either way, since rule is already providing options to have control over these behaviours.

Reply:
Oh wait, I think I got confused here - `ignoreInPromise` is the other way around, it's less strict if you set it to true. Maybe something like "allowPromiseMethods" would be a clearer option name?
Anyway, I think I'd favor `alwaysAwait: false, ignoreInPromise: false`. @SimenB should probably make the final call ^^

Reply:
Yeap, just to clarify. `ignoreInPromise: true` disables warnings for async assertions that were used in Promise methods. So setting it to `false` will complain for the usages below.

```js
// ignoreInPromise: false
// Bad
test("valid-expect", () => { 
  Promise.all([
    expect(Promise.resolve(2)).not.resolves.toBeDefined(),
    expect(Promise.resolve(3)).not.resolves.toBeDefined(),
  ]); 
});
```

```js
// ignoreInPromise: false
// Good
test("valid-expect", async () => { 
  await Promise.all([
    expect(Promise.resolve(2)).not.resolves.toBeDefined(),
    expect(Promise.resolve(3)).not.resolves.toBeDefined(),
  ]); 
});
```

```js
// ignoreInPromise: true
// Good
test("valid-expect", () => { 
  Promise.all([
    expect(Promise.resolve(2)).not.resolves.toBeDefined(),
    expect(Promise.resolve(3)).not.resolves.toBeDefined(),
  ]); 
});
```
I think when it's documented name of the options are fine but I'm ok with renaming options, of course.

Lastly, seems like tests on master are broken? Is it only me?